openff/toolkit/topology/topology.py

```diff
@@ -1557,7 +1589,13 @@ def to_openmm(self, ensure_unique_atom_names=True):
 
     @requires_package("openmm")
     def to_file(
```

Comment:
New functionality to test:
- [ ] `ensure_unique_atom_names` argument
    - [ ] `True` is previous behaviour
    - [ ] `False` skips uniquification
    - [ ] `str` is per hierarchy element
- [ ] Passing file-like object instead of path
    - [ ] Path still works
    - [ ] File like object works
- [ ] Optional positions

Reply:
There is a lot of repeated code in the existing tests for `to_file`. I don't want to change the existing functionality tests in a PR where I'm trying to avoid changing the functionality they test, but I think these tests could be improved (and reorganised) at a later date.

Reply:
Just recording a bug I found in the existing behavior while writing these tests: `Topology.to_openmm(ensure_unique_atom_names=False)` will raise an exception from the guts of OpenMM if any atom name is `None`, such as when a molecule is initialized from a SMILES. Might be frustrating for a user who is combining molecules from lots of different places and doesn't want to risk losing atom names. A simple fix might just be to call `str()` on the atom name before passing it to OpenMM, so that missing atom names are rendered in the PDB as the string "None" 😂

Reply:
Ah, that's a little obnoxious. I've opened an issue so we can handle it in a bugfix release.